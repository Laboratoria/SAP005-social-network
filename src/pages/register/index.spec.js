import { Register } from './index.js';
import * as services from '../../services/index.js';

describe('Register', () => {
  beforeEach(() => {
    services.createProfile = jest.fn(() => Promise.resolve(true));
    services.confirmEmail = jest.fn(() => Promise.resolve(true));
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should be a function', () => {
    expect(typeof Register).toBe('function');
  });
  it('shoul return registerPage', () => {
    expect(Register()).toMatchSnapshot();
  });
  it('should complete the register', async () => {
    const registerTest = Register();
    const btnRegister = registerTest.querySelector('#btnRegister');
    btnRegister.dispatchEvent(new Event('click'));
    expect(services.createProfile).toHaveBeenCalled();
  });
});
