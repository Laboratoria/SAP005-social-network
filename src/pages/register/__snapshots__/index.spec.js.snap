// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register shoul return registerPage 1`] = `
<div
  class="form-container"
>
  
  
  <div
    class="form-login"
  >
    
  
    <div
      class="header-container"
    >
      
    
      <figure
        class="logo"
      >
        <a
          href="/"
        >
          <img
            alt="Logo Runners - Voltar"
            id="logo"
            src="./assets/logo/runners-360px.png"
          />
        </a>
      </figure>
      
    
      <h1>
        Cadastre-se!
      </h1>
      
    
      <p
        class="about"
      >
        Compartilhe suas corridas, metas e desafios!
      </p>
      
    
      <hr />
      
  
    </div>
    
  
    <div>
      
    
      <p
        class="register"
      >
        Não compartilhamos seus dados com terceiros ;)
      </p>
      
  
    </div>
    
  
    <div
      class="row clearfix"
    >
      
    
      <div
        class="col_half"
      >
        
      
        <div
          class="input_field"
        >
          <span>
            <i
              aria-hidden="true"
              class="fas fa-running"
            />
          </span>
          
        
          <input
            id="displayName"
            placeholder="Nome"
            required=""
            type="text"
          />
                
      
        </div>
        
      
        <div
          class="input_field"
        >
          <span>
            <i
              aria-hidden="true"
              class="far fa-envelope"
            />
          </span>
          
        
          <input
            id="email"
            name="email"
            placeholder="Digite seu e-mail"
            required=""
            type="email"
          />
          
      
        </div>
        
    
      </div>
      
    
      <div
        class="col_half last"
      >
        
      
        <form>
          
        
          <h3
            class="error"
            id="msgError"
          />
          
        
          <div
            class="input_field"
          >
            <span>
              <i
                aria-hidden="true"
                class="fa fa-lock"
              />
            </span>
            
          
            <input
              id="password"
              placeholder="Senha"
              type="password"
            />
            
        
          </div>
          
        
          <div
            class="input_field"
          >
            <span>
              <i
                aria-hidden="true"
                class="fa fa-lock"
              />
            </span>
            
          
            <input
              id="confirmPwd"
              placeholder="Confirmar Senha"
              type="password"
            />
            
        
          </div>
          
      
        </form>
        
    
      </div>
      
  
    </div>
    

  </div>
  

  <div>
    <input
      class="button"
      id="btnRegister"
      type="submit"
      value="Cadastre-se"
    />
    

  </div>
  
  
</div>
`;

exports[`Register shoul return registerpag 1`] = `
<div
  class="form-container"
>
  
  
  <div
    class="form-login"
  >
    
  
    <div
      class="header-container"
    >
      
    
      <figure
        class="logo"
      >
        <a
          href="/"
        >
          <img
            alt="Logo Runners - Voltar"
            id="logo"
            src="./assets/logo/runners-360px.png"
          />
        </a>
      </figure>
      
    
      <h1>
        Cadastre-se!
      </h1>
      
    
      <p
        class="about"
      >
        Compartilhe suas corridas, metas e desafios!
      </p>
      
    
      <hr />
      
  
    </div>
    
  
    <div>
      
    
      <p
        class="register"
      >
        Não compartilhamos seus dados com terceiros ;)
      </p>
      
  
    </div>
    
  
    <div
      class="row clearfix"
    >
      
    
      <div
        class="col_half"
      >
        
      
        <div
          class="input_field"
        >
          <span>
            <i
              aria-hidden="true"
              class="fas fa-running"
            />
          </span>
          
        
          <input
            id="displayName"
            placeholder="Nome"
            required=""
            type="text"
          />
                
      
        </div>
        
      
        <div
          class="input_field"
        >
          <span>
            <i
              aria-hidden="true"
              class="far fa-envelope"
            />
          </span>
          
        
          <input
            id="email"
            name="email"
            placeholder="Digite seu e-mail"
            required=""
            type="email"
          />
          
      
        </div>
        
    
      </div>
      
    
      <div
        class="col_half last"
      >
        
      
        <form>
          
        
          <h3
            class="error"
            id="msgError"
          />
          
        
          <div
            class="input_field"
          >
            <span>
              <i
                aria-hidden="true"
                class="fa fa-lock"
              />
            </span>
            
          
            <input
              id="password"
              placeholder="Senha"
              type="password"
            />
            
        
          </div>
          
        
          <div
            class="input_field"
          >
            <span>
              <i
                aria-hidden="true"
                class="fa fa-lock"
              />
            </span>
            
          
            <input
              id="confirmPwd"
              placeholder="Confirmar Senha"
              type="password"
            />
            
        
          </div>
          
      
        </form>
        
    
      </div>
      
  
    </div>
    

  </div>
  

  <div>
    <input
      class="button"
      id="btnRegister"
      type="submit"
      value="Cadastre-se"
    />
  </div>
  
  
</div>
`;
